ecosystem/sep-0030.md

```diff
@@ -257,7 +258,10 @@ Name | Type | Description
 `identities` | array | A list of identities that if any one is authenticated will be able to gain control of this account.
 `identities[].role` | string | Same as request. See above.
 `identities[].authenticated` | boolean | Optional. Present and true if the currently authenticated client is authenticated with the identity.
-`signer` | string | The signer public key that the server will sign transactions with for this account when requested by an authenticated user.
+`signer` | string | The most recently added signer public key that the server will sign transactions with for this account when requested by an authenticated user.
```

Comment:
Alternatively, the server could query the account in the network, compare their signers with the ones it has available for it, and use the `signer` attribute to highlight the most recent available signer that has already been added to the account. But that's probably too complex.

Reply:
I think it's important to note that the signers provided by this API are never guaranteed to be signers of the account. They are keys that the server is willing to use to sign transactions for the account. I think `available_signers` has similar problems, it's easy to read that and think it is signers available on the account.

I think we should avoid querying the network here and really let the client choose the right key. There's more failure modes if we do this and it will be the easiest route for the client which means most clients will never actively choose a key and will just hope it works.

I'm all for removing `signer` and only having `signers`, I'll mark it as deprecated so that we keep it while we transition in the reference implementation and existing clients but will remove it soon. I think only having one will make this clearer. 

Does that work for you?

```suggestion
`signer` | string | **Deprecated.** The most recently added signer public key that the server will sign transactions with for this account when requested by an authenticated user.
```

Reply:
Here's a more drastic change that I think more completely addresses your concern: leighmcculloch/stellar--stellar-protocol#4. It's a breaking change though.

Reply:
Agreed that `available_signers` would convey the same wrong meaning, so I'm in favor of keeping `signers`. 

I like what you did in leighmcculloch#4 – is the plan to merge it at a later stage? It seems OK to make the breaking change (as this is a draft SEP), we just need to make sure we communicate with clients who are potentially relying on this field (if any) :)

Reply:
@fnando If we made the changes in leighmcculloch#4 would it be possible for your client code to test for one or the other, e.g. test for `signers[0].key || signer` and for us to have a brief period of overlap where we provide both fields? Also, do you use the `signer` field in the `/accounts/<address>/sign` endpoint?

Reply:
I think we could just keep the `signer` around for sometime, and maybe returning the full object in the signing endpoint doesn't really matter. I think I'll update this PR to reflect that conservative approach and we can continue to consider leighmcculloch#4 as the breaking change option.

Reply:
@accordeiro and I chatted offline yesterday and decided the renames in the current version of the PR and the plan to remove the confusing signer field completely in the next revision of this SEP are sufficient to address the confusion. Also, the issue still remains that a client needs to know which signer to use. To do this a client can either store the signer they expect to use in their own database, which Vega is likely to do, or the client can use Horizon to get a list of signers for the account and find the intersection of that list and the list the Recoverysigner returns and then choose the most recently added signing key that way. This change leaves open the opportunity for us to do that for the client in the Recoverysigner server if it turns out it was the wrong decision to offload this onto the client.